src/place-screen/hooks/use-departures-data.ts

```diff
@@ -296,33 +303,34 @@ export function useDeparturesData(
     return () => timeout.abort();
   }, [loadDepartures]);
   useEffect(() => {
-    if (!state.tick) {
-      return;
-    }
-    const diff = differenceInMinutes(state.tick, state.lastRefreshTime);
+    if (!state.tick) return;
+
+    const timeSinceLastHardRefresh = differenceInMinutes(
+      state.tick,
+      lastHardRefreshTime.current,
+    );
+    const timeSinceLastRealtimeRefresh = differenceInSeconds(
+      state.tick,
+      lastRealtimeRefreshTime.current,
+      // Rounding up makes ticks 10, 20 and 30s instead of 9, 19, and 29
+      {roundingMethod: 'ceil'},
+    );
 
-    if (diff >= HARD_REFRESH_LIMIT_IN_MINUTES) {
+    if (timeSinceLastHardRefresh >= HARD_REFRESH_LIMIT_IN_MINUTES) {
       loadDepartures();
+    } else if (timeSinceLastRealtimeRefresh >= updateFrequencyInSeconds) {
+      loadRealTimeData();
     }
-  }, [state.tick, state.lastRefreshTime]);
+  }, [state.tick]);
   useInterval(
-    loadRealTimeData,
-    updateFrequencyInSeconds * 1000,
-    [JSON.stringify(quayIds)],
-    !isFocused || mode === 'Favourite',
-  );
-  useInterval(
-    () => dispatch({type: 'TICK_TICK'}),
+    () => {
+      if (isFocused) dispatch({type: 'TICK_TICK'});
+    },
     tickRateInSeconds * 1000,
-    [],
+    [isFocused],
     !isFocused || mode === 'Favourite',
```

Comment:
Not sure what you mean here 🤔 `disabled` is set to

- `!isFocused` because we want to disable the loop when the view is not in focus.
- `mode === 'Favourite'` since the favorites flow doesn't use any updates.

Reply:
But if `isFocused` in the disabled param worked as intended, then we didn't need to check whether `isFocused` was true or not in the function body before dispatching the tick?

Reply:
Sorry about the (very) late reply here 😅 

If I removed the if statement, a tick was triggered when the screen loses focus. Turns out, there was a bug in `useInterval`: https://github.com/AtB-AS/mittatb-app/pull/3739